src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs

```diff
@@ -61,23 +62,19 @@ internal abstract class AbstractSnippetProvider : ISnippetProvider
         /// Determines if the location is valid for a snippet,
         /// if so, then it creates a SnippetData.
         /// </summary>
-        public async Task<SnippetData?> GetSnippetDataAsync(Document document, int position, CancellationToken cancellationToken)
+        public SnippetData? GetSnippetDataCore(SyntaxContext context, CancellationToken cancellationToken)
```

Comment:
i'd still just call this GetSnippetData.  Not sure why it needs to be called Core.